### PR TITLE
Defect Takedown: Sensu Check LRO Disabled

### PR DESCRIPTION
**Pull request by ddaskal**

### CODE REVIEW

**Comment by ddaskal** [comment]
This PR is also dependent on https://github.com/blueboxgroup/ursula-monitoring/pull/81

**Comment by ddaskal** [comment]
This PR Jenkins Test will fail if there are no bonded interfaces in CI environment.

File: roles/common/tasks/monitoring.yml [hunk 1]
```diff
@@ -191,9 +191,14 @@
...
+               args='--subnet {{ item.cidr }} --gateway {{ item.gateway }} --criticality {{ common.monitoring.sensu_checks.check_static_route.criticality }}'
   with_items: "{{ undercloud_cidr }}"
-  when: "{{undercloud_cidr is defined and undercloud_cidr | length > 1}}"
+  when: "{{ undercloud_cidr is defined and undercloud_cidr | length > 1 }}"
+  notify: restart sensu-client
+
+- name: check that large receive offload (LRO) is off
+  sensu_check: name=check-large-receive-offload plugin=check-large-receive-offload.py
+               args="--devices r\'"{{ hostvars[inventory_hostname].ansible_bond0.slaves }}"\' --criticality {{ common.monitoring.sensu_checks.check_large_receive_offload.criticality }}"
   notify: restart sensu-client
```
Comment (paulczar) [inline roles/common/tasks/monitoring.yml]: I think this should be gated off `when: common.monitoring.sensu_checks.check_lro.enabled|bool`

File: roles/common/tasks/monitoring.yml [hunk 2]
```diff
@@ -191,9 +191,14 @@
...
-               args='--subnet {{ item.cidr }} --gateway {{ item.gateway }} --criticality {{ common.monitoring.sensu_checks.check_static_route.criticality}}'
+               args='--subnet {{ item.cidr }} --gateway {{ item.gateway }} --criticality {{ common.monitoring.sensu_checks.check_static_route.criticality }}'
   with_items: "{{ undercloud_cidr }}"
-  when: "{{undercloud_cidr is defined and undercloud_cidr | length > 1}}"
+  when: "{{ undercloud_cidr is defined and undercloud_cidr | length > 1 }}"
+  notify: restart sensu-client
+
+- name: check that large receive offload (LRO) is off
+  sensu_check: name=check-large-receive-offload plugin=check-large-receive-offload.py
+               args="--devices r\'"{{ hostvars[inventory_hostname].ansible_bond0.slaves }}"\' --criticality {{ common.monitoring.sensu_checks.check_large_receive_offload.criticality }}"
```
Comment (paulczar) [inline roles/common/tasks/monitoring.yml]: this should probably just be a comma  maybe default to `eth0` and use join in the check ( probably got syntax wrong, i don't think its actually a filter ):

`common.monitoring.sensu_checks.check_lro.interfaces: []`

```
- name: check that large receive offload (LRO) is off
  sensu_check: name=check-large-receive-offload plugin=check-large-receive-offload.py
               args="--devices r\'"{{ common.monitoring.sensu_checks.check_lro.interfaces|join(',') }}"\' --criticality {{ common.monitoring.sensu_checks.check_large_receive_offload.criticality }}"
```

**Comment by notnownikki** [comment]
retest

File: roles/common/tasks/monitoring.yml (hunk 2 above)
Comment (j2sol) [inline roles/common/tasks/monitoring.yml]: Agreed, this will fail in places without a bond0 device.

**Comment by ddaskal** [comment]
I've addressed previous comments to make this automated for VM's or BM's based on the `primary_interface` of the environment. Once I get the +2, I'll squash commits. I thought I'd leave them separate so I have a place to fall back to if the latest changes are rejected.

**Comment by j2sol** [comment]
Okay, I think this is workable.

**Pull request merged**